Update JP block style consistency after core update

davemart-in opened the pull request:
## Description

One of the effects of the merge of [this core PR](https://github.com/WordPress/gutenberg/pull/18745) is that blank slate screens go from being center aligned to being left aligned.

@huguesvincent raised a flag when he noticed that many of the JP blocks now look off visually.

The goal of this PR is to reign in the blank slate screen design across all JP blocks.

## Testing Instructions

- Set up a local JP dev environment
- Make sure Gutenberg plugin is also installed
- Test each block listed in the change log below

## Change Log

Here's a detailed list of changes I'm made including before/after screenshots for each update:

### Contact Form

Before:

![](https://p111.p2.n0.cdn.getcloudapp.com/items/jkunAvl0/Image+2020-01-15+at+2.33.45+PM.png)

After:

![](https://p111.p2.n0.cdn.getcloudapp.com/items/Z4uwnDx1/Image+2020-01-15+at+3.14.59+PM.png)

Changes:

- Replaced top `<p>` tag with `placeholder__instructions` to reduce amount of padding between elements.
- Reduced bottom margin of `help-message`.
- Lightened up the `help-message` text to give everything a touch more visual hierarchy
- Adjusted margin around extra line of text and button


### Gif

Before:

![](https://p111.p2.n0.cdn.getcloudapp.com/items/nOumKbWR/Image+2020-01-15+at+3.17.12+PM.png)

After:

![](https://p111.p2.n0.cdn.getcloudapp.com/items/d5uegyvP/Image+2020-01-15+at+3.40.43+PM.png)

Changes:

- Removed `margin:0 auto;` which was forcing the input to be centered
- Removed isLarge CSS property to button
- Added isSecondary CSS property to button
- Adjusted height of input to match button height

Note:

I left a not on [this core issue](https://github.com/WordPress/gutenberg/issues/19662#issuecomment-575149675) to see if we can get the input and button height issue addressed in core. If that happens, I'll circle back and remove my extra input height CSS here.

### MailChimp

Before:

![](https://p111.p2.n0.cdn.getcloudapp.com/items/d5ueD8yd/Image+2020-01-16+at+9.54.42+AM.png)

After:

![](https://p111.p2.n0.cdn.getcloudapp.com/items/rRuLzXG9/Image+2020-01-16+at+10.14.17+AM.png)

Changes:

- Moved instructions to instructions prop
- Removed <br> tags

### Map

Before:

![](https://p111.p2.n0.cdn.getcloudapp.com/items/NQuvpqlW/Image+2020-01-16+at+10.17.45+AM.png)

After:

![](https://p111.p2.n0.cdn.getcloudapp.com/items/p9u5ZxAv/Image+2020-01-16+at+10.36.14+AM.png)

Changes:

- Moved instructions to instructions prop
- Removed <br> tags
- Added a bit more structure
- Added isSecondary to button

### Pinterest

Before:

![](https://p111.p2.n0.cdn.getcloudapp.com/items/WnuE6KZr/Image+2020-01-16+at+10.38.57+AM.png)

After:

![](https://p111.p2.n0.cdn.getcloudapp.com/items/v1umWR9b/Image+2020-01-16+at+10.40.53+AM.png)

Changes:

- Added isSecondary to button

### Recurring Payments

Before:

![](https://p111.p2.n0.cdn.getcloudapp.com/items/9ZuNy2pP/Image+2020-01-16+at+10.42.43+AM.png)

After:

![](https://p111.p2.n0.cdn.getcloudapp.com/items/mXumGE5X/Image+2020-01-16+at+11.14.52+AM.png)

Changes:

- Moved instructions to instructions prop
- Removed some unnecessary CSS

### Repeat Visitor

Before:

![](https://p111.p2.n0.cdn.getcloudapp.com/items/Z4uwGZGz/Image+2020-01-16+at+11.20.11+AM.png)

After:

![](https://p111.p2.n0.cdn.getcloudapp.com/items/YEud48nQ/Image+2020-01-16+at+11.25.26+AM.png)

Changes:

- Added margin between label and input
- Removed some unnecessary margin-bottom

Commit:
Updates to contact form block blank slate

Before: https://share.getcloudapp.com/jkunAvl0
After: https://share.getcloudapp.com/Z4uwnDx1

Commit:
Gif block blanks slate screen updates

Before: https://share.getcloudapp.com/nOumKbWR
After: https://share.getcloudapp.com/d5uegyvP

Commit:
Mailchimp block blank slate design update

Before: https://share.getcloudapp.com/d5ueD8yd
After: https://share.getcloudapp.com/rRuLzXG9

Commit:
Map block blank slate design improvements

Before: https://share.getcloudapp.com/NQuvpqlW
After: https://share.getcloudapp.com/p9u5ZxAv

Commit:
Pinterest block blank slate design update

Before: https://share.getcloudapp.com/WnuE6KZr
After: https://share.getcloudapp.com/v1umWR9b

Commit:
Recurring Payments blank slate screen one design

Before: https://share.getcloudapp.com/9ZuNy2pP
After: https://share.getcloudapp.com/mXumGE5X

Commit:
Repeat visitor block blank slate design updates

Before: https://share.getcloudapp.com/Z4uwGZGz
After: https://share.getcloudapp.com/YEud48nQ

Commit:
Repeat visitor block, reduced excess margin a bit more

folletto commented:
Design wise 👍

jeherve commented on extensions/blocks/gif/edit.js:
```diff
@@ -143,7 +143,7 @@ class GifEdit extends Component {
 					value={ searchText }
 				/>
-				<Button isLarge onClick={ this.onSubmit }>
+				<Button isSecondary onClick={ this.onSubmit }>
```
I think this is problematic on dark themes.

**Before**
![image](https://user-images.githubusercontent.com/426388/72548730-91740500-388f-11ea-8f1a-cf0fe286ee2e.png)

**After**

![image](https://user-images.githubusercontent.com/426388/72548717-8ae58d80-388f-11ea-9007-9a1991a68b27.png)

davemart-in replied:
Thanks! This should be fixed with: https://github.com/Automattic/jetpack/pull/14371#issuecomment-575345537

jeherve commented on extensions/blocks/mailchimp/edit.js:
```diff
@@ -149,23 +149,21 @@ class MailchimpSubscribeEdit extends Component {
-					<Button isDefault isLarge href={ connectURL } target="_blank">
-						{ __( 'Set up Mailchimp form', 'jetpack' ) }
-					</Button>
-					<br />
```
I think the line breaks were needed in some scenarios. In the example below, I am on a site that does not use the Gutenberg plugin:

![image](https://user-images.githubusercontent.com/426388/72548796-bcf6ef80-388f-11ea-95e4-1ff08dba6deb.png)

vs. before the patch:

![image](https://user-images.githubusercontent.com/426388/72548810-c718ee00-388f-11ea-9532-e648f240ccf5.png)

davemart-in replied:
Updated in https://github.com/Automattic/jetpack/pull/14371#issuecomment-575345537 👍

jeherve reviewed (changes requested):
I tested this on a site running WordPress 5.2 and no Gutenberg plugin, since we have to support those types of setup as well. I found a few issues, listed below.

I think it'd be nice if we could support both Gutenberg users (i.e. Atomic site owners) as well as the majority of Jetpack site owners that do not use the Gutenberg plugin.

Commit:
Added padding between MailChimp icon and name

Commit:
Moved className to placeholder on MailChimp block

Commit:
Added padding between gif block icon and title

Commit:
Visual updates to MailChimp block

- Put recheck connection button on new line
- Put max-width on information div

Commit:
Changes to form and gif block

- Removed information dive max-width
- Made buttons isDefault (note: for some reasons, isSecondary does not
work on either of these blocks)

Commit:
Updated MailChimp button placement and color of alert text in repeat visitor block

davemart-in commented:
Another round of updates:

**MailChimp**

- Added padding in-between block icon and title
- Made button isDefault (note: for some reasons, isSecondary does not work on this block button)
- I consulted @itsjonq and ended up having to change up the display property on the fields div in order to make the "Re-check Connection" link drop properly to the next line and work both with and without the Gutenberg plugin.

**Contact Form**

- Changed button from isPrimary to isSecondary to match other buttons
- Put max-width on information div

**Gif**

- Added padding in-between block icon and title
- Made button isDefault (note: for some reasons, isSecondary does not work on this block button)

**Repeat Visitor**

- I updated the alert box text since it was affected by the dark theme.

### Visual Preview

Here's a comparison of each of the blocks without Gutenberg, with Gutenberg, and with a dark theme:

![preview](https://user-images.githubusercontent.com/5634774/72562575-bfe6e580-3879-11ea-8b5c-7648ab8aec55.png)

scruffian commented on extensions/blocks/contact-form/components/jetpack-contact-form.js:
```diff
@@ -270,18 +263,21 @@ class JetpackContactForm extends Component {
+								</div>
 								<div className="jetpack-contact-form__create">
-									<Button isPrimary type="submit" disabled={ this.hasEmailError() }>
+									<Button isDefault type="submit" disabled={ this.hasEmailError() }>
```
Should this be isSecondary too? I guess we should only have one primary button per page?

davemart-in replied:
@scruffian as far as I can tell isDefault == isSecondary. For some reason isSecondary does not work on this block. If you have time to dig into why that is, that would be awesome. If not, no worries!

scruffian replied:
Sure, but let's do it on another PR

scruffian commented on extensions/blocks/contact-form/editor.scss:
```diff
@@ -20,14 +20,18 @@
 	}
+
+	.components-button {
+		margin: 6px 0 0;
```
This seems like a case of us overriding the default styles without good reason.  I think we can just remove it.

davemart-in replied:
Good call. Removed in https://github.com/Automattic/jetpack/commit/fe080710f57fdade89a28cab5dedf38525804336

scruffian commented on extensions/blocks/gif/editor.scss:
```diff
@@ -7,6 +7,9 @@
 		text-align: center;
 	}
+	.components-placeholder__label svg {
+		margin-right: 1ch;
```
I think this looks much better, but we should make the change in core, not here - all the blocks suffer from this lack of spacing.

davemart-in replied:
Happy to make this change in core, but I think we should keep it in here until that change has merged.

scruffian commented on extensions/blocks/gif/editor.scss:
```diff
@@ -27,17 +30,16 @@
 		width: 100%;
 		z-index: 1;
+		.components-text-control__input {
+			height: 36px;
```
This shouldn't be needed, but the GIF block is doing bad things with the placeholder and label, so I'll clean that up...

davemart-in replied:
Awesome. Thank you!

scruffian commented on extensions/blocks/mailchimp/editor.scss:
```diff
@@ -2,6 +2,10 @@
 .wp-block-jetpack-mailchimp {
 
+	.components-placeholder__label svg {
+		margin-right: 1ch;
```
We should fix this in core

davemart-in replied:
Ditto to above. Let's keep this hack in here until this is resolved in core.

scruffian committed:
add all instructions to the placeholder

Commit:
Remove needless button margin on form block

scruffian committed:
change TextControl to an input, so that it picks up the placeholder styles

scruffian committed:
Merge branch 'update/jp-block-style-consistency' of https://github.com/Automattic/jetpack into update/jp-block-style-consistency

jeherve reviewed (dismissed):
I think things look a lot better now. The only issue left seems to be with the GIF block, as discussed above.

Edit: annnd.. I just spotted @scruffian's changes as I was posting this :) I'll check again

scruffian committed:
remove unnecessary code

scruffian commented:
I made a few changes to the map block to remove some code we don't need:
<img width="474" alt="Screenshot 2020-01-17 at 11 04 32" src="https://user-images.githubusercontent.com/275961/72607924-7221ba00-3919-11ea-9797-60efe0a29240.png">

scruffian reviewed (commented):
I made a few comments, but this is looking much better :)

jeherve reviewed (approved):
We seem to be in a good spot right now; the GIF search button is still not perfect, but everything else is better than when we started.

folletto commented:
The changes happened from my initial design review above seem to have tweaked it a little more, so still all good 👍

jeherve commented:
r201783-wpcom
